pkg/controllers/provisioning/scheduling/machinetemplate.go

```diff
@@ -98,10 +96,7 @@ func (i *MachineTemplate) ToMachine(owner *v1alpha5.Provisioner) *v1alpha1.Machi
 		},
 	}
 	if i.Provider != nil {
-		raw := lo.Must(json.Marshal(i.Provider)) // Provider should already have been validated so this shouldn't fail
-		m.Annotations = lo.Assign(m.Annotations, map[string]string{
-			v1alpha5.ProviderCompatabilityAnnotationKey: string(raw),
-		})
+		m.Annotations = lo.Assign(m.Annotations, v1alpha5.ProviderAnnotation(i.Provider))
```

Comment:
optional: if you make `v1alpha5.ProviderAnnotation(i.Provider)` return `nil`, then you don't need the if on line 98.